coordinator/src/routes/logging.ts

```diff
@@ -0,0 +1,130 @@
+import {FhirPathBuilder, FhirPathReader} from "../../../models/common"
+import {fhir} from "../../../models"
+import {BundlePathBuilder} from "../../../models/common/fhir-path-builder/Bundle"
+import {ClaimPathBuilder} from "../../../models/common/fhir-path-builder/Claim"
+import {ParametersPathBuilder} from "../../../models/common/fhir-path-builder/Parameters"
+import {TaskPathBuilder} from "../../../models/common/fhir-path-builder/Task"
+import {
+  isBundle,
+  isClaim,
+  isParameters,
+  isTask
+} from "../utils/type-guards"
+
+const VALUE_NOT_PROVIDED = "NotProvided"
+
+type PayloadIdentifiers = {
+  payloadIdentifier: string
+  patientNhsNumber: string
+  senderOdsCode: string
+  prescriptionShortFormId: string
+}
+
+interface FhirPathGetter {
+  getPayloadIdentifier(): string
+  getNhsNumber(): string
+  getOdsCode(): string
+  getPrescriptionNumber(): string
+}
+
+type FhirPathBuilderTypes = BundlePathBuilder | ClaimPathBuilder | ParametersPathBuilder | TaskPathBuilder
+
+abstract class AbstractPathGetter<T extends FhirPathBuilderTypes> implements FhirPathGetter {
+  protected readonly builder: T
+
+  constructor(builder: T) {
+    this.builder = builder
+  }
+
+  getPayloadIdentifier(): string {
+    throw new Error("Method not implemented.")
+  }
+  getNhsNumber(): string {
+    throw new Error("Method not implemented.")
+  }
+  getOdsCode(): string {
+    throw new Error("Method not implemented.")
+  }
+  getPrescriptionNumber(): string {
+    throw new Error("Method not implemented.")
+  }
+}
+
+class BundlePathGetter extends AbstractPathGetter<BundlePathBuilder> {
+  constructor() {
+    super(new FhirPathBuilder().bundle())
+  }
+
+  getPayloadIdentifier = (): string => this.builder.identifier()
+  getNhsNumber = (): string => this.builder.patient().nhsNumber()
+  getOdsCode = (): string => this.builder.messageHeader().sender().identifier()
+  getPrescriptionNumber = (): string => this.builder.medicationRequest().prescriptionShortFormId()
+}
+
+class ClaimPathGetter extends AbstractPathGetter<ClaimPathBuilder> {
+  constructor() {
+    super(new FhirPathBuilder().claim())
+  }
+
+  getPayloadIdentifier = (): string => this.builder.identifier()
+  getNhsNumber = (): string => this.builder.patient().nhsNumber()
+  getPrescriptionNumber= (): string => this.builder.prescription().shortFormId()
+  getOdsCode = (): string => this.builder.organization().odsCode()
+}
+
+// TODO: Add examples for single patient and bulk release
```

Comment:
Hi @StefanoFrazzetto, have we got a ticket to cover this "TODO"?

Reply:
As discussed earlier, we might have more examples and/or have plans to add more end-to-end examples. Leaving the TODO for posterity, as a reminder for this.